spec/helpers/application_helper_spec.rb

```diff
@@ -36,4 +36,78 @@
       end
     end
   end
+
+  describe '#liveness_checking_enabled?' do
+    let(:liveness_checking_enabled) { false }
+    let(:sp_session) { {} }
+    let(:current_user) { nil }
+    before do
+      allow(FeatureManagement).to receive(:liveness_checking_enabled?).
+        and_return(liveness_checking_enabled)
+      allow(helper).to receive(:sp_session).and_return(sp_session)
+      allow(helper).to receive(:current_user).and_return(current_user)
+    end
+
+    context 'feature disabled' do
+      it 'returns false' do
+        expect(helper.liveness_checking_enabled?).to eq(false)
+      end
+    end
+
+    context 'feature enabled' do
+      let(:liveness_checking_enabled) { true }
+
+      context 'sp session value set' do
```

Comment:
this looks like an empty context? like it doesn't change anything? can we remove it and un-nest?

```suggestion
```

Reply:
Yeah, I'll unnest. Originally I think I was trying to capture the general behavior of "if the session value is set to anything at all, it must use that value", but I guess that's a rather open-box approach, when in reality we just want to be sure that `ial_strict: false` doesn't fall through to consider the user's past profile.

Reply:
Updated in 9293409.